pkg/providers/openai/openaiclient.go

```diff
@@ -0,0 +1,80 @@
+// TODO: Explore resource pooling
+// TODO: Optimize Type use
+// TODO: Explore Hertz TLS & resource pooling
+// OpenAI package provide a set of functions to interact with the OpenAI API.
+package openai
+
+import (
+	"errors"
+	"fmt"
+
+	"glide/pkg/providers"
+
+	"github.com/go-playground/validator/v10"
+)
+
+const (
+	providerName    = "openai"
+	providerVarPath = "/Users/max/code/Glide/pkg/providers/providerVars.yaml"
```

Comment:
These abs paths would not work anywhere else. I think you know that but just wanted to overcommunicate that